framework/linter/linter.go

```diff
@@ -1,344 +1,64 @@
 package linter
 
 import (
-	"go/ast"
 	"go/token"
 	"go/types"
 
-	"github.com/go-toolsmith/astfmt"
+	"github.com/go-critic/go-critic/linter"
 )
 
-// CheckerCollection provides additional information for a group of checkers.
-type CheckerCollection struct {
-	// URL is a link for a main source of information on the collection.
-	URL string
-}
-
-// AddChecker registers a new checker into a checkers pool.
-// Constructor is used to create a new checker instance.
-// Checker name (defined in CheckerInfo.Name) must be unique.
-//
-// CheckerInfo.Collection is automatically set to the coll (the receiver).
-//
-// If checker is never needed, for example if it is disabled,
-// constructor will not be called.
-func (coll *CheckerCollection) AddChecker(info *CheckerInfo, constructor func(*CheckerContext) (FileWalker, error)) {
-	if coll == nil {
-		panic("adding checker to a nil collection")
-	}
-	info.Collection = coll
-	addChecker(info, constructor)
-}
-
-// CheckerParam describes a single checker customizable parameter.
-type CheckerParam struct {
-	// Value holds parameter bound value.
-	// It might be overwritten by the integrating linter.
-	//
-	// Permitted types include:
-	//	- int
-	//	- bool
-	//	- string
-	Value interface{}
-
-	// Usage gives an overview about what parameter does.
-	Usage string
-}
-
-// CheckerParams holds all checker-specific parameters.
-//
-// Provides convenient access to the loosely typed underlying map.
-type CheckerParams map[string]*CheckerParam
-
-// Int lookups pname key in underlying map and type-asserts it to int.
-func (params CheckerParams) Int(pname string) int { return params[pname].Value.(int) }
+// Deprecated: use linter.UnknownType.
+var UnknownType = linter.UnknownType
```

Comment:
Have added alises for all `linter` declarations. `golangci-lint` still can be built without any changes.